File: Sitecore.Rocks.Templates/Engine/TemplateEngine.cs
using System.IO;
using HandlebarsDotNet;
using Sitecore.Rocks.Templates.Engine.Helpers;
using Sitecore.Rocks.Templates.Extensions;

namespace Sitecore.Rocks.Templates.Engine
{
    public class TemplateEngine : ITemplateEngine
    {
        public TemplateEngine()
        {
            Handlebars.RegisterHelper("camelCase",
                new WithFirstArgumentHelper("camelCase", s => s.ToCamelCase()).HelperFunction);
            Handlebars.RegisterHelper("pascalCase",
                new WithFirstArgumentHelper("pascalCase", s => s.ToPascalCase()).HelperFunction);
            Handlebars.RegisterHelper("literal",
                new WithFirstArgumentHelper("literal", s => s.ToLiteral()).HelperFunction);
            Handlebars.RegisterHelper("where", 
                new WhereHelper().BlockHelperFunction);
            Handlebars.RegisterHelper("withFirst", 
                new WithFirstHelper().BlockHelperFunction);

            //_compiler.RegisterTag(new NewGuidTag(), false);
            //_compiler.RegisterTag(new IfEqualTag(), false);
            //_compiler.RegisterTag(new ElseEqualTag(), false);
        }

        public string Render(string source, object data)
        {
            return Handlebars.Compile(source)(data);
        }

        public void RegisterPartial(string name, string partialSource)
        {
            using (var reader = new StringReader(partialSource))
            {
                var partialTemplate = Handlebars.Compile(reader);
                Handlebars.RegisterTemplate(name, partialTemplate);
            }
        }
    }
}

File: Sitecore.Rocks.Templates.FSharp/TemplateEngine.fs
module Sitecore.Rocks.Templates.FSharp.TemplateEngine

    open HandlebarsDotNet
    open System.IO
    
    let Compile (source:string) =
            
        Handlebars.Compile(source)

    let private RegisterTemplate  name partialTemplate =
        
        Handlebars.RegisterTemplate(name, partialTemplate)        
        Handlebars

    let RegisterPartial name partialSource =
        
        use reader = new StringReader(partialSource)                    
        let partialTemplate = Handlebars.Compile(reader)
        RegisterTemplate name partialTemplate  

        |> ignore